cli/tests/pcluster/utils/test_pcluster_utils.py

```diff
@@ -189,14 +204,142 @@ def test_get_stack_resources(boto3_stubber, resources, error_message):
         )
     mocked_requests = [
         MockedBoto3Request(
-            method="describe_stack_resources", response=response, expected_params={"StackName": FAKE_STACK_NAME}
+            method="describe_stack_resources",
+            response=response,
+            expected_params={"StackName": FAKE_STACK_NAME},
+            generate_error=error_message is not None,
         )
     ]
-    generate_errors = error_message is not None
-    boto3_stubber("cloudformation", mocked_requests, generate_errors=generate_errors)
+    boto3_stubber("cloudformation", mocked_requests)
     if error_message is None:
         assert_that(utils.get_stack_resources(FAKE_STACK_NAME)).is_equal_to(resources)
     else:
         with pytest.raises(SystemExit, match=response) as sysexit:
             utils.get_stack_resources(FAKE_STACK_NAME)
         assert_that(sysexit.value.code).is_not_equal_to(0)
+
+
+def test_retry_on_boto3_throttling(boto3_stubber, mocker):
+    sleep_mock = mocker.patch("pcluster.utils.time.sleep")
+    mocked_requests = [
+        MockedBoto3Request(
```

Comment:
What about defining a constant for this MockedBoto3Request since it is used in many places?
Like you did for `_generate_stack_event`

Reply:
we decided to leave as is since they are testing different methods with different boto3 calls and having some duplication here helps the readability of tests